stixcore/tmtc/parser.py

```diff
@@ -37,10 +38,130 @@ def parameter_dict_2_PacketData(cls, d):
+            a flattend copy
+        """
+        new_root = PacketData()
+        self._flatten(new_root)
+        return new_root
+
+    def _flatten(self, new_root):
+        for attr, value in self.__dict__.items():
+            if isinstance(value, PacketData):
+                value._flatten(new_root)
+            elif isinstance(value, list):
+                if attr == "NIXD0159":
+                    NIX00065 = []
+                    for index, list_elem in enumerate(value):
+                        if isinstance(list_elem, PacketData):
+                            c_v = list_elem.NIX00065
+                            if isinstance(c_v, int):
+                                NIX00065.append(c_v)
+                            elif isinstance(c_v, list) and len(c_v) == 2:
+                                NIX00065.append(int.from_bytes((c_v[0]+1).to_bytes(2, 'big')
+                                                + c_v[1].to_bytes(1, 'big'), 'big'))
+                            else:
+                                raise ValueError(f'Continuation bits value of {len(c_v)} \
+                                not allowed (0, 1, 2)')
+                        else:
+                            NIX00065.append(1)
+                    attr = "NIX00065"
+                    value = NIX00065
+
+                for index, list_elem in enumerate(value):
+                    if isinstance(list_elem, PacketData):
+                        list_elem._flatten(new_root)
+                    else:
+                        if hasattr(new_root, attr):
+                            old_val = getattr(new_root, attr)
+                            if not isinstance(old_val, list):
+                                old_val = [old_val, np.array(value)]
+                            else:
+                                old_val.append(np.array(value))
+                            setattr(new_root, attr, old_val)
+
+                        else:
+                            setattr(new_root, attr, np.array(value))
+                        break
+            else:
+                if hasattr(new_root, attr):
+                    # TODO check if this is possible
+                    raise Exception("add value not to repeater NIX")
+                else:
+                    setattr(new_root, attr, value)
+
+    def set(self, nix, value):
+        """Set the paremeter vale by the NIX name.
+
+        Parameters
+        ----------
+        nix : `str`
+            The NIX name of the parameter.
+        value : 'any'
+            The new value.
+        """
+        setattr(self, nix, value)
+
+    def get(self, nix):
+        """Get the parameter value for a given NIX name.
+
+        Parameters
+        ----------
+        nix : `str`
+            The NIX name of the parameter.
+
+        Returns
+        -------
+        'any'
+            The found value.
+        """
+        return getattr(self, nix)
+
+    def apply(self, nix, callback, args, addnix=None):
```

Comment:
you could add **kwargs here and pass them to the callback too if we need to might be useful later